clippy_lints/src/once_cell_lazy.rs

```diff
@@ -0,0 +1,277 @@
+use clippy_config::msrvs::Msrv;
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::visitors::for_each_expr;
+use clippy_utils::{def_path_def_ids, fn_def_id, match_def_path, path_def_id};
+use rustc_data_structures::fx::FxIndexMap;
+use rustc_errors::Applicability;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::def_id::DefId;
+use rustc_hir::{self as hir, BodyId, Expr, ExprKind, Item, ItemKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::impl_lint_pass;
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Lints when a `once_cell::sync::Lazy` type is declared as static variables.
+    ///
+    /// ### Why restrict this?
+    /// Such useage have been superseded by the `std::sync::LazyLock` type,
+    /// replacing them would reduce dependency of `once_cell` crate.
+    ///
+    /// ### Example
+    /// ```rust
+    /// use once_cell_lazy::once_cell_lazy;
+    /// use once_cell::sync::Lazy;
+    ///
+    /// once_cell_lazy! {
```

Comment:
😂 lol, Because I initially named this link `lazy_static` and later running `cargo dev rename_lint` to rename it, but it changes everything in the docs as well, I thought I already changed those back apparently not~

Also, thanks for the you time reviewing this, it's kinda rough as I didn't got much time polishing it.